### PR TITLE
[stdlib] assertion overloads for `List`

### PR DESCRIPTION
**Pull request by jjvraw**
Add `assert_equal` and `assert_not_equal` overload for `List` comparisons, a nice-to-have in a few places. Most probably a temporary solution under the current trait system.

### CODE REVIEW

File: stdlib/test/testing/test_assertion.mojo [hunk 1]
```diff
@@ -28,6 +28,28 @@ from builtin._location import _SourceLocation
...
+
+    try:
+        assert_equal(1, 0)
+    except e:
+        assert_true("test_assertion.mojo:43:21: AssertionError:" in str(e))
+
+    try:
+        assert_not_equal(0, 0)
+    except e:
+        assert_true("test_assertion.mojo:48:25: AssertionError:" in str(e))
```
Comment (jjvraw) [inline stdlib/test/testing/test_assertion.mojo:50]: Moved here to remove dependency on line numbers and future changes

**Comment by helehex** [comment]
Thanks, I just had to add something like this to one of my repos, except for `List[List[Int]]`. I can see this being useful for the time being. It would simplify tests in `/test_list.mojo`
One suggestion: add a TODO to remove this once we have associated types or more powerful conditional conformance that can infer conformance based on methods which use a inferred parameter to conditionally conform to the trait.

**Comment by helehex** [comment]
I wish someone would just add the str overload back in. It's easier to track a single TODO on an overload, than to track everywhere where you are unable to use a method because of lacking features.

**Review by helehex — APPROVED** [review]
Thanks, I just added something very similar to one of my repos, except for `List[List[Int]]`. I can see this being useful for the time being. It would simplify tests in `/test_list.mojo`

One suggestion:
Add a TODO to remove this once we have associated types or more powerful conditional conformance that can infer conformance based on methods which use a inferred parameter to conditionally conform to the trait.

**Comment by jjvraw** [comment]
Cool. There are two PRs atm that would benefit from this as well. 

> Add a TODO to remove...

There's a comment right above the other overloads, will modify that.

> repos, except for `List[List[Int]]`

I actually opened a discussion relevant to this issue, #3357, if you have any opinions :)

**Comment by helehex** [comment]
Oh yeah, i remember a comment in there somewhere, i guess my PR which removes the need for the SIMD overload hasn't gotten merged yet.

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**